server/src/main/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitor.java

```diff
@@ -75,20 +75,31 @@ public void onNewInfo(ClusterInfo clusterInfo) {
         logger.trace("processing new cluster info");
 
         final int numberOfNodes = clusterInfo.getNodeUsageStatsForThreadPools().size();
-        final Set<String> nodeIdsExceedingLatencyThreshold = Sets.newHashSetWithExpectedSize(numberOfNodes);
+        final Set<String> nodeIdsExceedingQueueLatencyThreshold = Sets.newHashSetWithExpectedSize(numberOfNodes);
+        final Set<String> nodeIdsBelowQueueLatencyThreshold = Sets.newHashSetWithExpectedSize(numberOfNodes);
         clusterInfo.getNodeUsageStatsForThreadPools().forEach((nodeId, usageStats) -> {
             final NodeUsageStatsForThreadPools.ThreadPoolUsageStats writeThreadPoolStats = usageStats.threadPoolUsageStatsMap()
                 .get(ThreadPool.Names.WRITE);
             assert writeThreadPoolStats != null : "Write thread pool is not publishing usage stats for node [" + nodeId + "]";
-            if (writeThreadPoolStats.maxThreadPoolQueueLatencyMillis() > writeLoadConstraintSettings.getQueueLatencyThreshold().millis()) {
-                nodeIdsExceedingLatencyThreshold.add(nodeId);
+            if (writeThreadPoolStats.maxThreadPoolQueueLatencyMillis() >= writeLoadConstraintSettings.getQueueLatencyThreshold().millis()) {
+                nodeIdsExceedingQueueLatencyThreshold.add(nodeId);
+            } else {
+                nodeIdsBelowQueueLatencyThreshold.add(nodeId);
             }
         });
 
-        if (nodeIdsExceedingLatencyThreshold.isEmpty()) {
+        if (nodeIdsExceedingQueueLatencyThreshold.isEmpty()) {
             logger.trace("No hot-spotting nodes detected");
             return;
         }
+        if (nodeIdsBelowQueueLatencyThreshold.isEmpty()) {
+            logger.debug(
+                """
+                    Nodes [{}] are above the queue latency threshold, but there are no nodes below the threshold. Cannot rebalance shards.""",
+                nodeSummary(nodeIdsExceedingQueueLatencyThreshold)
+            );
+            return;
+        }
```

Comment:
I was thinking about this one, would we not need to exclude search nodes from this consideration? (i.e. we need to see indexing nodes that are below threshold, because the search nodes will always be below the threshold). And in serverless at least there's always a search node.